src/main/scala/mesosphere/marathon/upgrade/ReadinessBehavior.scala

```diff
@@ -0,0 +1,122 @@
+package mesosphere.marathon.upgrade
+
+import akka.actor.{ ActorLogging, ActorRef, Actor }
+import mesosphere.marathon.core.readiness.{ ReadinessCheckExecutor, ReadinessCheckResult }
+import mesosphere.marathon.core.task.Task
+import mesosphere.marathon.core.task.tracker.TaskTracker
+import mesosphere.marathon.event.{ DeploymentStatus, HealthStatusChanged, MesosStatusUpdateEvent }
+import mesosphere.marathon.state.{ AppDefinition, PathId, Timestamp }
+import mesosphere.marathon.upgrade.DeploymentManager.ReadinessCheckUpdate
+import mesosphere.marathon.upgrade.ReadinessBehavior.ScheduleReadinessCheckFor
+import rx.lang.scala.Subscription
+
+/**
+  * ReadinessBehavior makes sure all tasks are healthy and ready depending on the app definition.
+  * Listens for TaskStatusUpdate events, HealthCheck events and ReadinessCheck events.
+  * If a task becomes ready, the taskIsReady hook is called.
+  *
+  * Assumptions:
+  *  - the actor is attached to the event stream for HealthStatusChanged and MesosStatusUpdateEvent
```

Comment:
why don't we make the `preStart` method of this trait subscribe to the events instead of requiring the subclasses to do this by themselves?